### DISCUSSION TITLE
TypeError: fetch failed when building the app in Next 13.4

### DISCUSSION THREAD

**Discussion by vitormarkis** [planetscale/discussion#507] [FAQs]
I'm using Planetscale database proxy to runs the database `pscale connect <db-name> <branch> --port 3309`

```.env
DATABASE_URL="mysql://root@localhost:3309/db-name"
```

It's working on dev mode, also `npx prisma studio` connects sucessfully, but when i try to build the app, in the pages where i fetch data using next async components, it throws this error for multiple pages:

### Example where I fetch data in async component
```tsx
export async function PostsSection({}: { session?: Session }) {
  const session = await getSession(headers().get("cookie") ?? "")
  const response = await fetch(`${api_endpoint}/api/posts`, {
    next: { revalidate: 10 },
  })
  const unparsedData = await response.json()
  const posts = z.array(postFeedSchema).parse(unparsedData)
// ...
```

### The error on building logs

```
...
[=== ] - info Generating static pages (7/8)(node:1591) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
node:internal/deps/undici/undici:11118
    Error.captureStackTrace(err, this);
          ^

TypeError: fetch failed
    at Object.fetch (node:internal/deps/undici/undici:11118:11)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: Error: connect ECONNREFUSED 127.0.0.1:3000
      at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1300:16)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    errno: -111,
    code: 'ECONNREFUSED',
    syscall: 'connect',
    address: '127.0.0.1',
    port: 3000
  }
}

Node.js v18.12.1
...
```

I assume its a error related to my PC, because the current commit on production works, but building locally it throws this error.

**Comment by orware** [comment]
Hello @vitormarkis,

You're likely right that there is some issue/difference on your local machine vs. the production environment.

Based on what was shared I'm not quite sure what that would be unfortunately.

A few thoughts I wonder about though is whether your production setup is still using `pscale connect` as I would expect it isn't and is instead using our public hostname along with generated username/password credentials for your branch. That difference could provide one explanation.

I also wonder if you are making use of our serverless driver in any way in your code, but if you are using Prisma that doesn't seem likely as support for our serverless driver isn't available there at this time as far as I'm aware. But if you were using the serverless driver I think it may not work together with the `pscale connect` option at this time so that would provide another potential explanation.

The only thing I can think of aside from those things would be an issue with the undici side in some way, but I don't have enough information to know that for sure.
